File: backend/srv/src/main/java/com/project/srv/service/DetailsService.java
package com.project.srv.service;

import com.project.srv.bean.DetailsBean;
import com.project.srv.dao.DetailsDao;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
public class DetailsService {
    // nombreparticipant tout seul
    @Autowired
    private DetailsDao detailsDao;

    public List<DetailsBean> findByNombreParticipants(int nombreParticipants) {
        return detailsDao.findByNombreParticipants(nombreParticipants);
    }

    public List<DetailsBean> findByPrixTotalActivite(double prixTotalActivite) {
        return detailsDao.findByPrixTotalActivite(prixTotalActivite);
    }

    @Transactional
    public void deleteByNombreParticipants(int nombreParticipants) {
        detailsDao.deleteByNombreParticipants(nombreParticipants);
    }

    @Transactional
    public void deleteByPrixTotalActivite(double prixTotalActivite) {
        detailsDao.deleteByPrixTotalActivite(prixTotalActivite);
    }

    // Vous pouvez ajouter d'autres méthodes de service personnalisées ici si nécessaire
}

File: frontend/react/services/DetailsService.ts
import axios from 'axios';
import { Details } from './types'; 

const API_URL = 'http://localhost:8080/Gestion-Vol/Details';

const findByNombreParticipants = async (nombreParticipants: number): Promise<Details[]> => {
    const response = await axios.get<Details[]>(`${API_URL}/participants/${nombreParticipants}`);
    return response.data;
};

const findByPrixTotalActivite = async (prixTotalActivite: number): Promise<Details[]> => {
    const response = await axios.get<Details[]>(`${API_URL}/prix/${prixTotalActivite}`);
    return response.data;
};

const deleteByNombreParticipants = async (nombreParticipants: number): Promise<void> => {
    await axios.delete(`${API_URL}/participants/${nombreParticipants}`);
};

const deleteByPrixTotalActivite = async (prixTotalActivite: number): Promise<void> => {
    await axios.delete(`${API_URL}/prix/${prixTotalActivite}`);
};

// Ajoutez d'autres méthodes de service web ici si nécessaire

export const detailsService = {
    findByNombreParticipants,
    findByPrixTotalActivite,
    deleteByNombreParticipants,
    deleteByPrixTotalActivite,
};
